.ci/e2eFleetServer.groovy

```diff
@@ -0,0 +1,186 @@
+        }
+        stage('Build Elastic Agent dependencies') {
+          options { skipDefaultCheckout() }
+          parallel {
+            stage('Build Fleet Server') {
+              options { skipDefaultCheckout() }
+              steps {
+                gitCheckout(basedir: BASE_DIR, branch: 'master', githubNotifyFirstTimeContributor: true, repo: "git@github.com:${env.ELASTIC_REPO}.git", credentialsId: env.JOB_GIT_CREDENTIALS)
+                dir("${BASE_DIR}") {
+                  withGoEnv(){
+                    sh(label: 'Build Fleet Server', script: "make release")
+                    sh(label: 'Copy binaries to Agent Drop path', script: 'cp build/distributions/* ${AGENT_DROP_PATH}')
+                  }
+                }
+              }
+            }
+            stage('Build Elastic Agent Dependencies') {
+              options { skipDefaultCheckout() }
+              steps {
+                gitCheckout(basedir: BEATS_BASE_DIR, branch: 'master', githubNotifyFirstTimeContributor: true, repo: "git@github.com:${env.BEATS_ELASTIC_REPO}.git", credentialsId: env.JOB_GIT_CREDENTIALS)
+                dir("${BEATS_BASE_DIR}/x-pack/elastic-agent") {
+                  withGoEnv(){
+                    sh(label: 'Build Fleet Server', script: 'DEV=true SNAPSHOT=true PLATFORMS="+all linux/amd64" mage package')
+                  }
+                }
+                dir("${BEATS_BASE_DIR}/x-pack") {
+                  sh(label:'Copy Filebeat binaries to Agent Drop path', script: 'cp filebeat/build/distributions/* ${AGENT_DROP_PATH}')
+                  sh(label:'Copy Heartbeat binaries to Agent Drop path', script: 'cp metricbeat/build/distributions/* ${AGENT_DROP_PATH}')
+                  sh(label:'Copy Metricbeat binaries to Agent Drop path', script: 'cp heartbeat/build/distributions/* ${AGENT_DROP_PATH}')
+                }
+              }
+            }
+          }
+        }
+        stage('Build Elastic Agent including Fleet Server') {
+          options { skipDefaultCheckout() }
+          steps {
+            dir("${BEATS_BASE_DIR}/x-pack/elastic-agent") {
+              withGoEnv(){
+                sh(label: 'Build Fleet Server', script: "AGENT_DROP_PATH='${env.AGENT_DROP_PATH}' DEV=true SNAPSHOT=true PLATFORMS='+all linux/amd64' mage package")
+              }
+            }
+          }
+        }
+        stage('Run E2E Tests') {
+          options { skipDefaultCheckout() }
+          steps {
+            gitCheckout(basedir: E2E_BASE_DIR, branch: "${env.E2E_BASE_BRANCH}", githubNotifyFirstTimeContributor: true, repo: 'git@github.com:elastic/e2e-testing.git', credentialsId: env.JOB_GIT_CREDENTIALS)
+            dockerLogin(secret: "${DOCKER_ELASTIC_SECRET}", registry: "${DOCKER_REGISTRY}")
+            dir("${E2E_BASE_DIR}") {
+              withGoEnv(){
+                withEnv(["BEATS_LOCAL_PATH='${env.BEATS_BASE_DIR}"]) {
+                  sh(label: 'Run E2E Tests', script: './.ci/scripts/fleet-tests.sh ')
```

Comment:
We should trigger here a build in the right branch for the e2e project, instead of running all tests in serie, but we would need a way to pass the generated binaries to the downstream job. But how?

The most feasible manner that I can think of is uploading the binaries to a GCP bucket, then we might need a new namespaced dir for fleet-server binaries, finally adapting the code to calculate the download path in consequence (we currently rely on Beats repo's SHA1 to get the bucket path). cc/ @elastic/observablt-robots